internal/manifest/compose/convert.go

```diff
@@ -8,50 +8,144 @@ import (
 	"github.com/deref/exo/internal/util/yamlutil"
 )
 
-func Import(r io.Reader) manifest.LoadResult {
-	procfile, err := Parse(r)
+type Loader struct {
+	// ProjectName is used as a prefix for the resources created by this importer.
+	ProjectName string
+}
+
+func (i *Loader) Load(r io.Reader) manifest.LoadResult {
+	composeProject, err := Parse(r)
 	if err != nil {
 		return manifest.LoadResult{Err: fmt.Errorf("parsing: %w", err)}
 	}
-	return Convert(procfile)
+	return i.convert(composeProject)
 }
 
-func Convert(project *Project) manifest.LoadResult {
+func (i *Loader) convert(project *Project) manifest.LoadResult {
 	res := manifest.LoadResult{
 		Manifest: &manifest.Manifest{},
 	}
-	for originalName, service := range project.Services {
+
+	// Since containers reference networks and volumes by their docker-compose name, but the
+	// Docker components will have a namespaced name, so we need to keep track of which
+	// volumes/components a service references.
+	networksByComposeName := map[string]string{}
+	volumesByComposeName := map[string]string{}
+
+	for originalName, volume := range project.Volumes {
 		name := manifest.MangleName(originalName)
-		if name != originalName {
+		if originalName != name {
 			res = res.AddRenameWarning(originalName, name)
 		}
+
+		v := volume.ToMap()
+		if _, ok := v["name"]; !ok {
+			v["name"] = i.prefixedName(name)
+		}
+		volumesByComposeName[originalName] = v["name"].(string)
```

Comment:
This can potentially panic. Would have been nice to modify the types to get some validation during unmarshalling.